### PR TITLE
Herculean/Atlas implants get a few additional BOULDER/Equipment related benefits. Boulders are more robust. Natural Athletics gains.

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Herculean/Atlas spinal implants allow you to break boulders to process them manually. Doing so also grants athletics experience.

They also increase the speed at which you throw objects, and also increase the range further. The Atlas version also allows you to reduce equipment movespeed penalties. (Fikou wanted me to do this for specifically modsuits, but I don't see any reason this can't apply to more than that, modsuits are just generally the slowest equipment you could wear)

Boulders have 30 throw force, are by default HUGE, but have no throw range at all and really weak tk throw range. You need a throwing booster to throw boulders around. (They do extra damage to SPECIAL mobs, like lavaland bosses, because it's funny to kill a god-like being with a huge rock while slow as shit)

You naturally get athletics experience by doing some actions. Fireman carrying, climbing tables/crates, climbing rope and climbing ladders give you a slow and steady stream of experience. Rope gives a larger amount if only because the process for doing so is destructive in some way (you have to get more climbing rope eventually, so there is an associated cost). Spinal implants now improve climbing speed for climbing over objects.

## Why It's Good For The Game

I just had this strange compulsion to make the spinal implant make you good at throwing boulders at lethal speeds. It already has some cool rock-based special effects around you. Having mastery over rock throwing just kind of made sense. Additionally, I thought it would be a funny method for someone to be able to kill a boss via way of thrown boulders. Or some hapless nuclear operative unprepared for a bunch of assistants lobbing boulders at them.

Additionally, I also think it letting you break boulders for ore is a fairly minor but fun addition. Pull boulders out of the vents, smash them. Good work out. Particularly for miners.

I talked a while ago with I think Jacq about adding some natural methods for accumulating athletics experience over the course of a round that doesn't involve the dedicated weight lifting machinery/boxing bag. I thought any method for doing that should probably be relatively slow and definitely nonviable for significant athletics gains beyond novice level. Mooost people will hit novice on a multi-z map if they go up and down ladders enough. Or climb a lot of tables/crates. The results aren't going to have much noticeable impact, however, as you only gain appreciable benefits above novice. Probably just greater visibility of fitness in a round. (Rather nice for the RD, who gets to show off how swole they are more regularly).

<details>
  <summary>Spoiler warning</summary>

CES8b4XWEAAIakb

</details>

## Changelog
:cl:
balance: Herculean/Atlas spinal implants make you better at climbing tables, throw objects much faster and throw them much further. You can also break boulders with your bare hands. 
balance: Atlas implants let you offset the weight of worn/carried equipment (like boulders).
balance: Boulders have no throw range, but if you can throw them via throwing range boosters (such as the spinal implant), they deal fairly large amounts of damage. Especially to powerful mobs like lavlaand bosses.
balance: You naturally gain a small amount of athletics experience doing some tasks, such as climbing rope, ladders and tables/crates, fireman carrying, or smashing open boulders.
/:cl:

### CODE REVIEW

**Comment by Donglesplonge** [comment]
can more accurately larp as chris redfield now, speedmerge?

**Comment by necromanceranne** [comment]
New Athletics reward that is just a mug that says 'Don't Bother Me Until I've Punched My Morning Boulder'

**Comment by EnterTheJake** [comment]
Anne trying to make one bad PR challenge (impossible).

**Comment by Sealed101** [comment]
does Atlas also reduce the xenobio adamantium armor slowdown

**Comment by necromanceranne** [comment]
> does Atlas also reduce the xenobio adamantium armor slowdown

No....kinda. I don't know if it is intended to work like this, but anything marked as immutably slow applies the slowdown twice. Once for immutable, and once for normal equipment slow. These movespeed modifiers aren't mutally exclusive to one another (This is something that can be set). TRAIT_STURDY_FRAME alters the normal slow still, but it cannot alter the immutable slow. So it kind of shaves the slowdown to just above half the usual slowdown? I guess?

**Comment by Sealed101** [comment]
eh good enough

**Comment by Eronymun** [comment]
It reducing the slow down on adamantium would actually be a unique benefit over stabilized red. Other than that though they seem fairly similar in practice, and both are from sci, just different parts. Redundancy isn't a bad thing tho.

**Comment by necromanceranne** [comment]
I mean, it's not a huge deal, I make the regular argument that stabilized red probably shouldn't be letting you ignore equipment slowdown in its entirety but nobody is really moving to resolve that and I certainly am not going to do that myself. The Atlas has plenty else to entice people towards it. It really is just a cherry on top kind of benefit (particularly since it's an anomalock and it can get away with being a bit overloaded)

**Comment by CCC23** [comment]
> > does Atlas also reduce the xenobio adamantium armor slowdown
> 
> No....kinda. I don't know if it is intended to work like this, but anything marked as immutably slow applies the slowdown twice. Once for immutable, and once for normal equipment slow. These movespeed modifiers aren't mutally exclusive to one another (This is something that can be set). TRAIT_STURDY_FRAME alters the normal slow still, but it cannot alter the immutable slow. So it kind of shaves the slowdown to just above half the usual slowdown? I guess?

beautiful I now dont have to fully remove the slowdown with zero g and can now just be chunky wall man chucking boulders at nerds

**Comment by Loafin34** [comment]
https://github.com/user-attachments/assets/b4d5c2ae-8f05-4f45-a2ad-d12674c1397d

**Comment by DrAmazing343** [comment]
I had a vision of a Cyclopes of old Greek myth throwing boulders reading this. It was cerebral. speedmerge immediately.

**Comment by carlarctg** [comment]
kinda feel like strongarm should be the boulder breaker not herculean


also scope creep this to make syndie climbing hooks small sized, seriously why are they normal

also what if you could put the hercules implant inside modsuits to temporarily get its effects

**Comment by necromanceranne** [comment]
> kinda feel like strongarm should be the boulder breaker not herculean

I have something else in mind for that. Strongarm feels like a combat implant, and this is more a utilitarian one. Boulder breaking just feels like a utilitarian option to me.

> also scope creep this to make syndie climbing hooks small sized, seriously why are they normal

I mean thats a good suggestion but I'm probably not gonna do that in this PR. Good first PR for consistency changes.

> also what if you could put the hercules implant inside modsuits to temporarily get its effects

Maybe chat with fikou about that.

File: code/modules/surgery/organs/internal/cyberimp/augments_chest.dm [hunk 1]
```diff
@@ -305,7 +311,8 @@
 	if(istype(tool, /obj/item/assembly/signaler/anomaly/grav))
 		user.balloon_alert(user, "core installed.")
 		athletics_boost_multiplier = 0.25
-		added_throw_range += 2
+		added_throw_speed += 4
```
Comment (Fikou) [inline code/modules/surgery/organs/internal/cyberimp/augments_chest.dm:314]: have you tested this? +6 throw speed to any item seems a bit insane

Comment (necromanceranne) [reply code/modules/surgery/organs/internal/cyberimp/augments_chest.dm:314]: I tested it out. The effect that anything that could embed, will embed, and do a pretty decent amount of damage at that. It lets you embed glass shards for one.

For an anomaly core item, it didn't seem TOO extreme. The actual embedding weapons that we have don't meaningfully become too much more or less lethal using it from my testing.

**Comment by necromanceranne** [comment]
Boulders don't embed. This is to stop just turning glass shards into shuriken without Atlas.

**Comment by carlarctg** [comment]
content removal

**Comment by CCC23** [comment]
> Boulders don't embed. This is to stop just turning glass shards into shuriken without Atlas.

Put spikey tape on boulder and chuck it into the nearest feds chest

**Comment by necromanceranne** [comment]
I mean I guess.

**Comment by necromanceranne** [comment]
oh nooo spurious failures

**Comment by Eronymun** [comment]
> > Boulders don't embed. This is to stop just turning glass shards into shuriken without Atlas.
> 
> Put spikey tape on boulder and chuck it into the nearest feds chest

getting a boulder stuck to you sounds like you'd be stuck crawling at sisyphean speeds

**Review by Fikou — CHANGES_REQUESTED** [review]

File: code/datums/elements/climbable.dm [hunk 2]
```diff
@@ -64,6 +64,15 @@
 								span_notice("You start climbing onto [climbed_thing]..."))
 	var/adjusted_climb_time = climb_time
 	var/adjusted_climb_stun = climb_stun
+
+	var/fitness_level = user.mind?.get_skill_level(/datum/skill/athletics) - 1
+	adjusted_climb_time -= fitness_level
```
[the remark this answers is not in the document]
Comment (necromanceranne) [reply code/datums/elements/climbable.dm:69]: oh good point, I overlooked that.

File: code/modules/surgery/organs/internal/cyberimp/augments_chest.dm [hunk 3]
```diff
@@ -305,13 +311,29 @@
 	if(istype(tool, /obj/item/assembly/signaler/anomaly/grav))
 		user.balloon_alert(user, "core installed.")
 		athletics_boost_multiplier = 0.25
-		added_throw_range += 2
+		added_throw_speed += 5
+		added_throw_range += 4
 		strength_bonus += 4
 		core_applied = TRUE
 		name = "\improper Atlas gravitonic spinal implant"
```
[the remark this answers is not in the document]
Comment (necromanceranne) [reply code/modules/surgery/organs/internal/cyberimp/augments_chest.dm:318]: This is a newish thing right, how is that formatted? Just like that?

[the remark this answers is not in the document]
Comment (necromanceranne) [reply code/modules/surgery/organs/internal/cyberimp/augments_chest.dm:318]: 👍

File: code/datums/elements/climbable.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (Fikou) [reply code/datums/elements/climbable.dm:69]: also a comment about this being deciseconds please

**Comment by CCC23** [comment]
boulder time

**Pull request merged**